src/components/pages/request-holiday/request-holiday.component.js

```diff
@@ -18,32 +19,69 @@ import Section from '../../commons/section/section.component';
 
 import useStyles from './request-holiday.styles';
 
-const names = [
-  'Tom',
-  'Anna',
-  'Daniel',
-  'Cailey',
-];
-const RequestHoliday = () => {
+const initialState = {
+  holidayType: { value: 1, isValid: true },
+  notes: { value: '', isValid: true },
+  dateFrom: { value: moment(), isValid: true },
+  dateTo: { value: moment(), isValid: true },
+  requestedSupervisors: { value: [], isValid: true }
+};
+const reducer = (state, {
+  type, payload
+}) => {
+  console.log(type, payload);
+  switch (type) {
+    case 'DATE_FROM':
+      return { ...state, dateFrom: payload };
+    case 'DATE_TO':
+      return { ...state, dateTo: payload };
+    default:
+      return { ...state, [payload.name]: payload };
+  }
+};
+const RequestHoliday = ({ supervisors }) => {
   const classes = useStyles();
-  const [holidayType, setHolidayType] = React.useState(1);
-  const [sendRequestTo, setSendRequestTo] = React.useState({ value: [], isValid: true });
-  const [selectedDateFrom, handleDateChangeFrom] = React.useState(moment().valueOf());
-  const [selectedDateTo, handleDateChangeTo] = React.useState(moment().valueOf());
-  const [note, handleNotesChange] = React.useState('');
+
+  const [state, dispatchModification] = useReducer(reducer, initialState);
   useEffect(() => {
-    if (selectedDateFrom > selectedDateTo) handleDateChangeTo(selectedDateFrom);
-  }, [selectedDateFrom]);
+    if (state.dateFrom.value > state.dateTo.value) dispatchModification({ ...state.dateFrom, type: 'DATE_TO' });
+  }, [state.dateFrom]);
+
   const shouldDisableDate = day => (day.isoWeekday() === 6 || day.isoWeekday() === 7);
+  const handleChange = ({ target }) => { dispatchModification({ payload: target, isValid: true }); };
+  const handleDateChange = name => date => {
+    dispatchModification({ type: name, value: date, isValid: true });
+  };
+  const validateForm = formData => {
+    let isFormValid = true;
+    if (formData.requestedSupervisors.value.length === 0) {
+      isFormValid = false;
+      dispatchModification({
+        type: 'CHANGE_STATE',
+        stateToChange: 'requestedSupervisors',
+        value: { value: state.requestedSupervisors.value, isValid: false }
+      });
+    }
+    if (!formData.dateFrom.value) {
+      dispatchModification({ type: 'DATE_FROM', value: '', isValid: false });
+      isFormValid = false;
+    }
+    if (!formData.dateTo.value) {
+      dispatchModification({ type: 'DATE_TO', value: '', isValid: false });
+      isFormValid = false;
+    }
+    return isFormValid;
```

Comment:
This `let` based declaration can be changed, I suggest to iterate over the state values with the help of `Object.keys` and map.